app/models/attendance.rb

```diff
@@ -32,23 +34,94 @@ class Attendance < UuidApplicationRecord
     where('check_in BETWEEN ? AND ?', week.at_beginning_of_week(:sunday), week.at_end_of_week(:saturday))
   }
 
+  scope :absences, -> { where.not(absence: nil) }
+  scope :non_absences, -> { where(absence: nil) }
+
   scope :illinois_part_days, -> { where('total_time_in_care < ?', '5 hours') }
   scope :illinois_full_days, -> { where('total_time_in_care BETWEEN ? AND ?', '5 hours', '12 hours') }
   scope :illinois_full_plus_part_days, -> { where('total_time_in_care > ? AND total_time_in_care < ?', '12 hours', '17 hours') }
   scope :illinois_full_plus_full_days, -> { where('total_time_in_care BETWEEN ? AND ?', '17 hours', '24 hours') }
 
+  delegate :business, to: :child_approval
+  delegate :user, to: :child_approval
+  delegate :child, to: :child_approval
+  delegate :state, to: :child
+  delegate :county, to: :child
+  delegate :timezone, to: :user
+
   private
 
+  def round_check_in
```

Comment:
We're rounding the check-in and check-out to the just-passed minute by subtracting the seconds.